doc/cli/cli_dev.rst

```diff
@@ -0,0 +1,84 @@
+
+
+Extending the CLI
+=================
+
+Developers can create their own 'subcommands' for the ``pydicom`` command,
+by adding entry points to their package's setup.py file, specifying a callback
+function to register the subcommand and its arguments.
+
+If you wanted to create two subcommands, 'command1' and 'command2',
+you setup.py file should include something like:
+
+.. code-block:: python
+
+    if __name__ == '__main__':
+        setup(
+            name="yourpackage",
+            # various setup options...,
+            entry_points = {
+                "pydicom_subcommands": [
+                    "command1 = yourpackage.command1module.add_subparser",
+                    "command2 = yourpackage.command2module.add_subparser"
+                ]
+            }
+        )
+
+
+The ``"pydicom_subcommands"`` is a literal string; this must not be 
+changed or *pydicom* will not find your subcommand.
+
+The ``add_subparser`` function name could be changed if you wish, but usually
+would be used by convention, and is assumed in the following examples.
+
+In the module you have specified, create the ``add_subparser`` function,
+which takes a single argument ``subparsers``, and a ``do_command`` function,
+which will take the call when you subcommand is actually used at the command
+line::
+
+    from pydicom.cli.main import filespec_help, filespec_parser
+
+    def add_subparser(subparsers):
```

Comment:
An actual docstring and/or type hints would be helpful

Reply:
Yes, true -- I'll add some here and in most places once I finalize the 'API'.